PythonVisualizations/Array.py

```diff
@@ -224,44 +261,141 @@ def find(self, val):
                         posShape,
                         multiply_vector((1, 1, -1, -1), self.CELL_BORDER)),
                     outline=self.FOUND_COLOR))
-                self.wait(0.2)
-                
+
+                # update the display
+                self.window.update()
+                self.wait(0.1)
+
+                # Animation stops
+                self.highlightCodeTags([], callEnviron)
                 self.cleanUp(callEnviron)
                 return i
 
             # if not found, then move the index over one cell
+            self.highlightCodeTags('outer_loop_increment', callEnviron)
             self.moveItemsBy(indexDisplay, (self.CELL_SIZE, 0), sleepTime=0.01)
+            if self.wait(0.1):
+                break
+
+        # key not found
+        self.highlightCodeTags('key_not_found', callEnviron)
+        self.window.update()
+        self.wait(0.1)
 
+        # Animation stops
+        self.highlightCodeTags([], callEnviron)
         self.cleanUp(callEnviron)
         return None
 
+    removeCode = """
+def delete(self, item):
+    for j in range(self.nItems):
+        if self.__a[j] == item:
+            for k in range(j, self.nItems):
+                self.__a[k] = self.__a[k+1]
+            self.nItems -= 1
+            return True
+    return False

```

Comment:
This code doesn't quite match what's in the textbook.  The decrement of `nItems` has to happen before the inner for loop, not afterwards.  See Array.py in [02_code](https://drive.google.com/drive/folders/1OXk62CRfU6tsTvxXKUUpIHPUioT5u5U2)